### PR TITLE
Breaking: Refactor DOM utils inside hint to `@hint/utils`

### LINKED ISSUE #2192: Refactor DOM utils inside hint to `@hint/utils`
These should be able to be moved through a combination of moving any types they need to `@hint/utils` as well. The few exceptions are:
1. Use of `HTMLElement` in `content-type`
    * Replace with a minimal local interface
2. Use of `Engine` in `traverse`
    * Replace with `EventEmitter2`

Another thing to do is to re-export the types from `hint`.
Also classes like `HTMLDocument` might be moved as well.

### PR DESCRIPTION
**Pull request by sarvaje**
Fix #2192

## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/webhintio/hint)
- [x] Followed the [commit message guidelines](https://webhint.io/docs/contributor-guide/getting-started/pull-requests/#commit-messages)

For non-trivial changes, please make sure you also:

- [x] Added/Updated related documentation.
- [x] Added/Updated related tests.

### CODE REVIEW

**Comment by sarvaje** [comment]
@molant I have added documentation, can you review it again?

File: packages/connector-jsdom/src/connector.ts [hunk 1]
```diff
@@ -44,7 +44,9 @@ import { beforeParse } from './before-parse';
  * ------------------------------------------------------------------------------
  */
 
-const { dom: { createHTMLDocument, traverse }, contentType: { getContentTypeData, getType } } = utils;
+const { createHTMLDocument, traverse } = dom;
+
```
Comment (antross) [inline packages/connector-jsdom/src/connector.ts]:
```suggestion
```

File: packages/connector-local/src/connector.ts [hunk 2]
```diff
@@ -42,37 +42,33 @@ const readFileAsBuffer = promisify(readFile);
...
 
 /*
  * ------------------------------------------------------------------------------
  * Defaults
  * ------------------------------------------------------------------------------
  */
 
+const { traverse } = dom;
+const { contentType: { getContentTypeData, getType, isTextMediaType } } = utils;
+
```
Comment (antross) [inline packages/connector-local/src/connector.ts]:
```suggestion
```

**Comment by sarvaje** [comment]
I have accepted the suggested changes :)

**Pull request merged**